### ISSUE TITLE
torch.lu_unpack crash with segmentation fault in the nightly version

### ISSUE THREAD

**Issue by DNXie** [pytorch/pytorch#85026]
### 🐛 Describe the bug

torch.lu_unpack crash with segmentation fault in the nightly version

Both of the following inputs crash with segfault

```
import torch
print(torch.__version__)    #  1.13.0.dev20220913+cpu
torch.lu_unpack(LU_pivots=torch.ones((16,0), dtype=torch.int32), LU_data=torch.ones((0,20,4,12)))  
```

```
import torch
print(torch.__version__)    #  1.13.0.dev20220913+cpu
torch.lu_unpack(LU_pivots=torch.tensor([-10000000], dtype=torch.int32), LU_data=torch.tensor([[1]]))   # segfault
```

Also reproduced in this [gist](https://colab.research.google.com/drive/15KE8CmAhA6Kyf4-ciYOSBl_QWA0yWe5G?usp=sharing)

### Versions

Collecting environment information...
PyTorch version: 1.13.0.dev20220913+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.4 LTS (x86_64)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.10

Python version: 3.7.6 (default, Jan  8 2020, 19:59:22)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-4.15.0-176-generic-x86_64-with-debian-buster-sid
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] torch==1.13.0.dev20220913+cpu
[pip3] torchaudio==0.13.0.dev20220913+cpu
[pip3] torchvision==0.14.0.dev20220913+cpu
[conda] blas                      1.0                         mkl  
[conda] mkl                       2020.2                      256  
[conda] mkl-service               2.3.0            py37he8ac12f_0  
[conda] mkl_fft                   1.3.0            py37h54f3939_0  
[conda] mkl_random                1.1.1            py37h0573a6f_0  
[conda] numpy                     1.19.2           py37h54aff64_0  
[conda] numpy-base                1.19.2           py37hfa32c7d_0  
[conda] torch                     1.13.0.dev20220913+cpu          pypi_0    pypi
[conda] torchaudio                0.13.0.dev20220913+cpu          pypi_0    pypi
[conda] torchvision               0.14.0.dev20220913+cpu          pypi_0    pypi

cc @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano

**Comment by Lezcano** [comment]
Note that `lu_unpack` is expected to be used with the output of `linalg.lu_factor`. We cannot reliably check that the inputs come from a different function. We could put some guards in the relevant kernels to check that the pivots are not out of bounds.

Note that we have this issue reported a few times, but it is always reported by people using fuzzing tools to test the PyTorch API, it has never come from an actual usage. WDYT @mruberry?

**Comment by mruberry** [comment]
`module: edge cases` seems like the appropriate label. We could think about being a little more explicit in the documentation of `lu_unpack`, but it's pretty clear that it expects input from `lu_factor()`
